settings.gradle.kts

```diff
@@ -1,12 +1,84 @@
-include("atomicfu")
-include("atomicfu-transformer")
-include("atomicfu-gradle-plugin")
-include("atomicfu-maven-plugin")
+pluginManagement {
+    repositories {
+        repositories {
+            mavenCentral()
+            gradlePluginPortal()
+            mavenLocal()
 
-include("atomicfu-native") // this is a kludge so that existing YouTrack config works, todo: remove
-include("integration-testing")
+            val additionalRepositoryProperty = providers.gradleProperty("community.project.kotlin.repo")
+                .orElse(providers.gradleProperty("kotlin_repo_url"))
+            if (additionalRepositoryProperty.isPresent) {
+                maven(url = uri(additionalRepositoryProperty.get()))
+                logger.info("A custom Kotlin repository ${additionalRepositoryProperty.get()} was added")
+            }
+
+            /*
+             * This property group is used to build kotlinx.atomicfu against Kotlin compiler snapshots.
+             * When build_snapshot_train is set to true, kotlin_version property is overridden with kotlin_snapshot_version.
+             * Additionally, mavenLocal and Sonatype snapshots are added to repository list
+             * (the former is required for AFU and public, the latter is required for compiler snapshots).
+             * DO NOT change the name of these properties without adapting kotlinx.train build chain.
+             */
+            val buildSnapshotTrainGradleProperty = providers.gradleProperty("build_snapshot_train")
+            if (buildSnapshotTrainGradleProperty.isPresent) {
+                maven(url = uri("https://oss.sonatype.org/content/repositories/snapshots"))
+            }
+
+        }
+    }
+}
 
 dependencyResolutionManagement {
+    val buildSnapshotTrainGradleProperty = providers.gradleProperty("build_snapshot_train")
+
+    @Suppress("UnstableApiUsage")
+    repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)
+
+    @Suppress("UnstableApiUsage")
+    repositories {
+
+        // TODO(Dmitrii Krasnov): maybe could be passed via additionalRepositoryProperty
+        maven("https://maven.pkg.jetbrains.space/kotlin/p/kotlin/dev")
```

Comment:
I requested a list of TeamCity CI builds, to check which configurations use this repository

Reply:
The same about `mavenLocal()`